app/controllers/experiments_controller.rb

```diff
@@ -687,6 +698,38 @@ def results_binaries
                                              @experiment.id, @user_session)
   end
 
+  def notifications
+    notifications = MongoActiveRecord.get_collection('notifications')
+    tailf = Mongo::Cursor.new(notifications, tailable: true, order: [['$natural', 1]])
+
+    response.headers['Content-Type'] = 'text/event-stream'
+    sse = SSE.new(response.stream)
+
+    begin
+      loop do
+        while (doc = tailf.next) != nil
+
+          if doc['experiment_id'].to_s == @experiment.id.to_s and
+              %w(progress-bar-update progress-bar-refresh simulation-started simulation-progress-update
+                 simulation-stopped simulation-completed simulation-manager-update moes).include?(doc['event'])
+            sse.write(doc, event: doc['event'])
+          end
+        end
+
+        sleep 1
+      end
+
+      render nothing: true
+
+    rescue IOError
+      logger.debug("IO error")
```

Comment:
Should be "Experiments controller notifications IO Error: #{e.to_s}" to facilitate debugging.